scripts/dns-test-execute.bash

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env bash
+
+# The integration tests call this to test Linux DNS control, using the `/etc/resolv.conf`
+# method which only works well inside Alpine Docker containers.
+
+set -euo pipefail
+
+HTTPBIN=test.httpbin.docker.local
+
+# Wait for client to ping httpbin (CIDR) resource through the gateway
+docker compose exec -it client timeout 60 \
+sh -c "until ping -W 1 -c 10 172.20.0.100 &>/dev/null; do true; done"
+
+echo "# check original resolv.conf"
+docker compose exec -it client sh -c "cat /etc/resolv.conf.firezone-backup"
+
+echo "# Make sure gateway can reach httpbin by DNS"
+docker compose exec -it gateway sh -c "curl $HTTPBIN/get"
+
+echo "# Try to ping httpbin as a DNS resource"
+docker compose exec -it client timeout 60 \
+sh -c "ping -W 1 -c 10 $HTTPBIN"
+
+echo "# Access httpbin by DNS"
+docker compose exec -it client sh -c "curl $HTTPBIN/get"
+
+echo "# Make sure it's going through the tunnel"
+docker compose exec -it client timeout 60 \
+sh -c "nslookup $HTTPBIN; nslookup $HTTPBIN | tail -n +4 | grep 100\\.96\\.0\\."
+
+echo "# Make sure a non-resource doesn't go through the tunnel"
+docker compose exec -it client timeout 60 \
+sh -c "nslookup github.com; nslookup github.com | tail -n +4 | grep -v 100\\.96.\\0\\."
```

Comment:
Why do we need to look it up twice? I think if you want to see the output, `tee` could be your friend too.

Reply:
Once so it shows up in the CI logs for manual debugging, then once for the test.

`tee` doesn't seem to have a flag for copying to stderr, so I could either copy to a temp file and print that out, or use this syntax from SO: `command | tee >(cat 1>&2)`

If I wrap that in a function it's not too bad. I'll try it.